src/main/java/com/google/growpod/servlets/UserServlet.java

```diff
@@ -120,6 +120,53 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         HttpServletResponse.SC_METHOD_NOT_ALLOWED, "Unimplemented: " + request.getRequestURI());
   }
 
+  /**
+   * Processes HTTP POST requests for the /user servlet. Dispatches functionality based on structure
+   * of POST request.
+   *
+   * @param request Information about the POST Request
+   * @param response Information about the servlet's response
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /* uriList will have "" as element 0 */
+    String[] uriList = request.getRequestURI().split("/");
+    assert (uriList[1].equals("user"));
+
+    if (uriList.length < 3) {
```

Comment:
Should this be moved above the assert in L134? It's possible `uriList[1]` does not exist.

Reply:
The first and second elements of `uriList` should always be `"/"` and `"user"` if this particular method is executed. If this is not the case, then there is something wrong with the servlet setup, and it would be inappropriate to return a 405 like your suggested change would.

Though I will add a length checking clause to the assertion.